sentence_transformers/evaluation/BinaryClassificationEvaluator.py

```diff
@@ -144,25 +149,31 @@ def __call__(self, model, output_path: str = None, epoch: int = -1, steps: int =
         return main_score
 
     def compute_metrices(self, model):
-        try:
-            # If the sentences are hashable, then we can use a set to avoid embedding the same sentences multiple times
-            sentences = list(set(self.sentences1 + self.sentences2))
-            embeddings = model.encode(
-                sentences, batch_size=self.batch_size, show_progress_bar=self.show_progress_bar, convert_to_numpy=True
-            )
-            emb_dict = {sent: emb for sent, emb in zip(sentences, embeddings)}
-            embeddings1 = [emb_dict[sent] for sent in self.sentences1]
-            embeddings2 = [emb_dict[sent] for sent in self.sentences2]
-        except TypeError:
-            # Otherwise we just embed everything, e.g. if the sentences are images for evaluating a CLIP model
-            embeddings = model.encode(
-                self.sentences1 + self.sentences2,
-                batch_size=self.batch_size,
-                show_progress_bar=self.show_progress_bar,
-                convert_to_numpy=True,
-            )
-            embeddings1 = embeddings[: len(self.sentences1)]
-            embeddings2 = embeddings[len(self.sentences1) :]
+        with nullcontext() if self.truncate_dim is None else model.truncate_sentence_embeddings(self.truncate_dim):
```

Comment:
Could instead do:

```python
with model.truncate_sentence_embeddings(self.truncate_dim or model.truncate_dim):
    ...
```

I made the current choice b/c then the `model` doesn't have to have a `truncate_sentence_embeddings` method for the evaluator to work when `self.truncate_dim is None` (b/c the last part of the ternary is never evaluated by Python). It is a little more to read though, so I'm open to changing it to the style above

Reply:
I like the `nullcontext` approach, that's how I would have done it too.